src/Core/Cart/Calculator.php

```diff
@@ -192,6 +208,20 @@ public function getDiscountTotal()
             $amount = $amount->add($cartRule->getDiscountApplied());
         }
 
+        $totalWithoutDiscount = $this->getRowTotalWithoutDiscount();
+
+        if (null !== $this->getFees()->getInitialShippingFees() && null !== $this->getFees()->getFinalShippingFees()) {
+            $shippingDiscount = (new AmountImmutable())
+                ->add($this->getFees()->getInitialShippingFees())
+                ->sub($this->getFees()->getFinalShippingFees())
+            ;
+            $totalWithoutDiscount = $totalWithoutDiscount->add($shippingDiscount);
```

Comment:
Shouldn't it be a substraction here since it's a discount?

Reply:
Or maybe it should be added to `$amount` ? To have the addition of product discount plus shipping discount?
By the way does `$totalWithoutDiscount` already contains shipping fees?

Reply:
Actually it's the rest of the shipping discount so it should be added. Maybe I should rename the `$shippingDiscount`to better reflect what this variable is?

Reply:
As seen with you on Slack, I changed the variable name to better understand that part